fsspec/implementations/tests/test_http.py

```diff
@@ -129,6 +132,17 @@ def test_list_cache_with_skip_instance_cache(server):
     assert out == [server + "/index/realfile"]
 
 
+def test_glob_return_subfolders(stdlib_simple_http_server):
+    h = fsspec.filesystem("http")
+    out = h.glob(stdlib_simple_http_server + "/*")
+    assert out == [
+        stdlib_simple_http_server + "/file1",
+        stdlib_simple_http_server + "/file2",
+        stdlib_simple_http_server + "/folder1/",
```

Comment:
This is a "folder" because it ends in "/" or because it contains further things? I think this is the crux of the original issue - in most filesystems, a path cannot end in "/" (the folder's actual path is without the slash).

Reply:
Your comment made me realize that I was not entirely aware of what is and is not a directory for http filesystems.

It's a folder because it contains further things. Or if I understand correctly: It's a folder because for a request to http<em></em>://server/somepath/<strong>folder1/</strong> the server returns html that contains links other than "." and ".."

Reply:
> It's a folder because for a request to http://server/somepath/folder1/ the server returns html that contains links other than "." and ".."

Yes, or more precisely links with the _same_prefix_ , like
- http://server/folder is a folder if it contains at least one link to
  - http://server/folder/...

As you say, we exclude relative links "." and ".." but also links to completely different paths.

Reply:
Given this discussion, do you still wish to continue with this PR?

Reply:
Sorry, I accidentally marked this as completed.

I'll make the changes and will update this PR when I find some free time.